From 7648a497bd1df4e69e7fa972cd58957e9e230429 Mon Sep 17 00:00:00 2001
From: David Runge <dvzrv@archlinux.org>
Date: Thu, 20 Feb 2025 09:10:33 +0100
Subject: [PATCH] python-linux-procfs: Remove use of the six Python library

Python 2 has been discontinued for several years.
The "six" Python library has been used to allow use of Python2 and
Python3 in the transitional period.
Remove the use of the library and exclusively rely on Python3 specific
calls to builtins.

Signed-off-by: David Runge <dvzrv@archlinux.org>
---
 pflags            | 1 -
 procfs/procfs.py  | 1 -
 procfs/utilist.py | 2 --
 setup.py          | 1 -
 4 files changed, 5 deletions(-)

diff --git a/pflags b/pflags
index 46d396c..4fde7ad 100755
--- a/pflags
+++ b/pflags
@@ -18,7 +18,6 @@
 import procfs, re, fnmatch, sys
 import argparse
 from functools import reduce
-from six.moves import map
 
 ps = None
 
diff --git a/procfs/procfs.py b/procfs/procfs.py
index 7cc7371..3fcc45c 100755
--- a/procfs/procfs.py
+++ b/procfs/procfs.py
@@ -11,7 +11,6 @@ import platform
 import re
 import time
 from functools import reduce
-from six.moves import range
 from procfs.utilist import bitmasklist
 
 VERSION = "0.7.3"
diff --git a/procfs/utilist.py b/procfs/utilist.py
index e6314f0..0ecf15d 100755
--- a/procfs/utilist.py
+++ b/procfs/utilist.py
@@ -6,8 +6,6 @@
 # Copyright (C) 2007 Red Hat, Inc.
 #
 
-from six.moves import range
-
 
 def hexbitmask(l, nr_entries):
     hexbitmask = []
diff --git a/setup.py b/setup.py
index 144e07e..6e268fb 100755
--- a/setup.py
+++ b/setup.py
@@ -29,5 +29,4 @@ Abstractions to extract information from the Linux kernel /proc files.
 """,
     packages = ["procfs"],
     scripts = ['pflags'],
-    install_requires = ['six'],
 )
-- 
2.48.1

